spec/chewy/search/request_spec.rb

```diff
@@ -0,0 +1,556 @@
+
+    specify { expect(described_class.new(ProductsIndex).limit(10)).to eq(described_class.new(ProductsIndex).limit(10)) }
+    specify { expect(described_class.new(ProductsIndex).limit(10)).not_to eq(described_class.new(ProductsIndex).limit(20)) }
+
+    specify { expect(ProductsIndex.limit(10)).to eq(ProductsIndex.limit(10)) }
+    specify { expect(ProductsIndex.limit(10)).not_to eq(CitiesIndex.limit(10)) }
+  end
+
+  describe '#render' do
+    specify do
+      expect(subject.render)
+        .to match(
+          index: %w[products],
+          type: array_including(%w[product city country])
+        )
+    end
+  end
+
+  describe '#inspect' do
+    specify do
+      expect(described_class.new(ProductsIndex).inspect)
+        .to eq('<Chewy::Search::Request {:index=>["products"], :type=>["product", "city", "country"]}>')
+    end
+    specify do
+      expect(ProductsIndex.limit(10).inspect)
+        .to eq('<ProductsIndex::Query {:index=>["products"], :type=>["product", "city", "country"], :body=>{:size=>10}}>')
+    end
+  end
+
+  %i[query post_filter].each do |name|
+    describe "##{name}" do
+      specify { expect(subject.send(name, match: {foo: 'bar'}).render[:body]).to include(name => {match: {foo: 'bar'}}) }
+      specify { expect(subject.send(name) { match foo: 'bar' }.render[:body]).to include(name => {match: {foo: 'bar'}}) }
+      specify do
+        expect(subject.send(name, match: {foo: 'bar'}).send(name) { multi_match foo: 'bar' }.render[:body])
+          .to include(name => {bool: {must: [{match: {foo: 'bar'}}, {multi_match: {foo: 'bar'}}]}})
+      end
+      specify { expect { subject.send(name, match: {foo: 'bar'}) }.not_to change { subject.render } }
+      specify do
+        expect(subject.send(name).should(match: {foo: 'bar'}).send(name).must_not { multi_match foo: 'bar' }.render[:body])
+          .to include(name => {bool: {should: {match: {foo: 'bar'}}, must_not: {multi_match: {foo: 'bar'}}}})
+      end
+
+      context do
+        let(:other_scope) { subject.send(name).should { multi_match foo: 'bar' }.send(name) { match foo: 'bar' } }
+
+        specify do
+          expect(subject.send(name).not(other_scope).render[:body])
+            .to include(name => {bool: {must_not: {bool: {must: {match: {foo: 'bar'}}, should: {multi_match: {foo: 'bar'}}}}}})
+        end
+      end
+    end
+  end
+
+  describe '#filter' do
+    specify { expect(subject.filter(match: {foo: 'bar'}).render[:body]).to include(query: {bool: {filter: {match: {foo: 'bar'}}}}) }
+    specify { expect(subject.filter { match foo: 'bar' }.render[:body]).to include(query: {bool: {filter: {match: {foo: 'bar'}}}}) }
+    specify do
+      expect(subject.filter(match: {foo: 'bar'}).filter { multi_match foo: 'bar' }.render[:body])
+        .to include(query: {bool: {filter: {bool: {must: [{match: {foo: 'bar'}}, {multi_match: {foo: 'bar'}}]}}}})
```

Comment:
Wouldn't it be better if it would render:
```
query: {bool: {filter: [{match: {foo: 'bar'}}, {multi_match: {foo: 'bar'}}]}}
```

it would have exact same behavior. So, if there is a filter with `must` clause only (no matter how many clauses are there) should render like this (similar to 2 examples above)